### ISSUE TITLE
frontend build error with octicons.js

### ISSUE THREAD

**Issue by tujian** [nteract/commuter#237]
```
➜  commuter git:(master) node --version
v8.9.0
➜  commuter git:(master) npm --version
5.5.1
➜  commuter git:(master) git rev-parse HEAD
484d26e83d5d1a25001c0d21566c39b8f49cea88
```

`npm run dev`
``` 
 DONE  Compiled successfully in 1998ms                                                                                16:08:11

[Wed Nov 01 2017 16:08:11 GMT+0800 (CST)] INFO Node env: development
[Wed Nov 01 2017 16:08:11 GMT+0800 (CST)] INFO Commuter server listening on port 4000
> Building page: /


 DONE  Compiled successfully in 1189ms                                                                                16:08:22

::1 - - [01/Nov/2017:08:08:22 +0000] "GET / HTTP/1.1" 302 -
> Building page: /view


 ERROR  Failed to compile with 1 errors                                                                               16:08:28                                                                          

 error  in ../frontend/icons/octicons.js

Module build failed: TypeError: /home/*/Repos/github/nteract/commuter/packages/frontend/icons/octicons.js: Cannot read property '0' of null
    at Buffer._append (/home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/buffer.js:95:25)
    at Buffer.append (/home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/buffer.js:72:10)
    at Generator._append (/home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/printer.js:206:52)
    at Generator.word (/home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/printer.js:131:10)
    at Generator.Identifier (/home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/generators/types.js:38:8)
    at /home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/printer.js:298:23
    at Buffer.withSource (/home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/buffer.js:168:5)
    at Generator.withSource (/home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/printer.js:189:15)
    at Generator.print (/home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/printer.js:297:10)
    at Generator.MemberExpression (/home/*/Repos/github/nteract/commuter/packages/frontend/node_modules/babel-generator/lib/generators/expressions.js:216:8)

 @ ../frontend/components/contents/directory-listing.js 19:16-47
 @ ../frontend/components/contents/index.js
 @ ../frontend/pages/view.js?entry
 @ multi ../frontend/pages/view.js?entry
```

**Comment by tujian** [comment]
# Problem solved after I fix package version of `next` to `4.0.0-beta.2`

## patch
```diff
diff --git a/packages/frontend/package.json b/packages/frontend/package.json
index dc84ad7..74062be 100644
--- a/packages/frontend/package.json
+++ b/packages/frontend/package.json
@@ -46,7 +46,7 @@
     "isomorphic-fetch": "^2.2.1",
     "jquery": "^3.2.1",
     "mathjax-electron": "^2.0.1",
-    "next": "^4.0.0-beta.2",
+    "next": "4.0.0-beta.2",
     "nprogress": "^0.2.0",
     "octicons": "^6.0.1",
     "prop-types": "^15.5.10",
```

## before patch
```
➜  frontend git:(master) ✗ npm list babel-generator
@nteract/commuter-frontend@2.0.0 /Users/arthur/Repos/github/nteract/commuter/packages/frontend
└─┬ next@4.1.4
  ├─┬ babel-core@6.26.0
  │ └── babel-generator@6.26.0  deduped
  └── babel-generator@6.26.0
```

## after patch
```
➜  frontend git:(master) ✗ npm list babel-generator
@nteract/commuter-frontend@2.0.0 /Users/arthur/Repos/github/nteract/commuter/packages/frontend
└─┬ next@4.0.0-beta.2
  ├─┬ babel-core@6.26.0
  │ └── babel-generator@6.26.0  deduped
  └── babel-generator@6.26.0
```

**Comment by rgbkrk** [comment]
Oh hey we should bring commuter up to date as is, want to send a PR to bring it up to `4.1.4`?

**Comment by tujian** [comment]
Of course I would like to submit a clean fix very much.  However, I have been away from node.js and babel stuffs for a long long time (considering the fast changes of node.js stacks).

Today I came across this error for the first time and realized debugging it was beyond my scope. I made a guess that it was caused by compatibility breaking because no one ever reported this error before, and lucky I am : )

Nevertheless, I did find out related issues on another repo (gaearon/react-hot-loader#689, gaearon/react-hot-loader#686). I have subscribed these issues and will try to understand the problem and hopefully submit a PR here.

If anyone already have a solution, feel free to submit it.
